File: Examples/VisualBasic/AsposePDF/Pages/InsertEmptyPageAtEnd.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf

Namespace AsposePDF.Pages
    Public Class InsertEmptyPageAtEnd
        Public Shared Sub Run()
            ' ExStart:InsertEmptyPageAtEnd
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Pages()

            ' Open document
            Dim pdfDocument1 As New Document(dataDir & Convert.ToString("InsertEmptyPageAtEnd.pdf"))

            ' Insert an empty page at the end of a PDF file
            pdfDocument1.Pages.Add()

            dataDir = dataDir & Convert.ToString("InsertEmptyPageAtEnd_out_.pdf")
            ' Save output file
            pdfDocument1.Save(dataDir)
            ' ExEnd:InsertEmptyPageAtEnd
            System.Console.WriteLine(Convert.ToString(vbLf & "Empty page inserted successfully at the end of document." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Pages/InsertEmptyPageAtEnd.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Pages
{
    public class InsertEmptyPageAtEnd
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Pages();

            //open document
            Document pdfDocument1 = new Document(dataDir + "InsertEmptyPageAtEnd.pdf");

            //insert an empty page at the end of a PDF file
            pdfDocument1.Pages.Add();

            //save output file
            pdfDocument1.Save(dataDir + "InsertEmptyPageAtEnd_out.pdf");

        }
    }
}